### PR TITLE
Train : TrainDisappearChange introduced

### PR DESCRIPTION
**Pull request by bgiuliana**

### CODE REVIEW

File: src/main/java/fr/sncf/osrd/train/events/TrainCreatedEvent.java [hunk 1]
```diff
@@ -52,17 +52,17 @@ public boolean deepEquals(TimelineEvent other) {
...
         return event;
     }
 
-    public static class TrainCreationPlanned extends Simulation.TimelineEventCreated {
+    public static class TrainPlannedCreation extends Simulation.TimelineEventCreated {
         public final TrainSchedule schedule;
 
         /** Plans the creation of some train */
-        private TrainCreationPlanned(Simulation sim, TrainSchedule schedule) {
+        private TrainPlannedCreation(Simulation sim, TrainSchedule schedule) {
```
Comment (eckter) [inline src/main/java/fr/sncf/osrd/train/events/TrainCreatedEvent.java:65]: Is there a reason why this is renamed?

File: src/test/java/fr/sncf/osrd/train/DisappearTrainTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+import static fr.sncf.osrd.Helpers.*;
+import static fr.sncf.osrd.train.TestTrains.FAST_NO_FRICTION_TRAIN;
+import static org.junit.jupiter.api.Assertions.*;
+
+
+public class DisappearTrainTest {
+
+    @Test
+    public void simpleOccupationTest() throws InvalidInfraException, SimulationError, InvalidSchedule {
```
Comment (eckter) [inline src/test/java/fr/sncf/osrd/train/DisappearTrainTest.java]: The test name should describe what is tested

File: src/test/java/fr/sncf/osrd/train/DisappearTrainTest.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        TrainCreatedEvent.plan(sim, schedule2);
+
+        // run the simulation
+        while (!sim.isSimulationOver())
+            sim.step();
+
+        var lastPosition1 = sim.trains.get("train1").getLastState().location.getPathPosition();
+        var lastPosition2 = sim.trains.get("train2").getLastState().location.getPathPosition();
+
+        assertEquals(lastPosition1, lastPosition2, 10);
```
Comment (eckter) [inline src/test/java/fr/sncf/osrd/train/DisappearTrainTest.java:129]: You can also test that the train status are REACHED_DESTINATION

File: src/test/java/fr/sncf/osrd/train/StopTests.java [hunk 4]
```diff
@@ -66,6 +67,47 @@ public void testStopDuration() throws InvalidInfraException {
...
+                new RJSTrainStop[]{
+                        new RJSTrainStop(1000., null, 0),
+                        new RJSTrainStop(-1., null, 0)
+                }
+        );
+
+        var sim = Simulation.createFromInfraAndEmptySuccessions(RailJSONParser.parse(infra), 0, null);
+        run(sim, config);
+
+        assertTrue(sim.isSimulationOver());
```
Comment (eckter) [inline src/test/java/fr/sncf/osrd/train/StopTests.java]: This doesn't really test anything, the simulation is always over after `run()`. You could test that the train last speed and fail if it's below a certain value.

File: src/main/java/fr/sncf/osrd/train/events/TrainCreatedEvent.java (hunk 1 above)
Comment (bgiuliana) [reply src/main/java/fr/sncf/osrd/train/events/TrainCreatedEvent.java:65]: The same of TrainPlannedMove, TrainPlannedRestart, TrainPlannedMoveToActionPoint

**Pull request merged**